package NYT.NTableClient.NProto;

import "yt/ytlib/chunk_holder/chunk.proto";
import "yt/ytlib/table_client/schema.proto";

////////////////////////////////////////////////////////////////////////////////

message TBlockInfo
{
    required int32 block_index = 1;
    required int32 row_count = 2;
}

message TChunkChannel
{
    required TChannel channel = 1;
    repeated TBlockInfo blocks = 2;
}

message TTableChunkAttributes
{
    extend NYT.NChunkHolder.NProto.TChunkAttributes
    {
        required TTableChunkAttributes table_attributes = 101;
    }

    repeated TChunkChannel chunk_channels = 1;
    required int32 codec_id = 2;
    required int64 row_count = 3;

    // ToDo: make required.
    optional int64 uncompressed_size = 4;
}

////////////////////////////////////////////////////////////////////////////////

